DistributedKansa.ps1

```diff
@@ -0,0 +1,538 @@
+    [Parameter(Mandatory=$false,Position=12)]
+        [String[]]$ElkAlert=@(),
+    [Parameter(Mandatory=$false,Position=13)]
+        [String]$KansaLocalPath="$($PSScriptRoot)",
+    [Parameter(Mandatory=$false,Position=14)]
+        [String]$KansaRemotePath="F:\Kansa",
+    [Parameter(Mandatory=$False,Position=15)]
+        [int]$JobTimeout=600,
+    [Parameter(Mandatory=$false,Position=16)]
+        [int[]]$ElkPort=@(13651),
+    [Parameter(Mandatory=$false,Position=17)]
+        [Switch]$FireForget,
+    [Parameter(Mandatory=$false,Position=18)]
+        [String]$FFwrapper="Modules\FireForget\FFwrapper.ps1",
+    [Parameter(Mandatory=$false,Position=19)]
+        [hashtable]$FFArgs=@{delayedStart = $true; maxDelay = 3600; killSwitch = $true; killDelay = 1800; VDIcheck = $false; CPUpriority = "Idle" },
+    [Parameter(Mandatory=$false,Position=20)]
+        [String]$FFStagePath="Modules\FFStaging\",
+    [Parameter(Mandatory=$false,Position=21)]
+        [Switch]$Overwrite,
+    [Parameter(Mandatory=$false,Position=22)]
+        [String]$SafeWord="safeword",
+    [Parameter(Mandatory=$false,Position=23)]
+        [Switch]$noPrompt
+)
+
+function Validate-Creds{
+    param(
+    [Parameter(Mandatory=$True,Position=0)]
+        [System.Management.Automation.PSCredential]$cred
+    )
+    $username = $cred.username
+    $password = $cred.GetNetworkCredential().password
+
+    # Get current domain using logged-on user's credentials
+    $CurrentDomain = "LDAP://" + ([ADSI]"").distinguishedName
+    $domain = New-Object System.DirectoryServices.DirectoryEntry($CurrentDomain,$UserName,$Password)
+
+    if ($domain.name -eq $null)
+    {
+     Write-Warning "Authentication failed - please verify your username and password."
+     exit #terminate the script.
+    }
+}
+
+function Get-Modules {
+<#
+.SYNOPSIS
+Looks for modules.conf in the $Modulepath, default is Modules. If found,
+returns an ordered hashtable of script files and their arguments, if any. 
+If no modules.conf is found, returns an ordered hashtable of all modules
+found in $Modulepath, but no arguments will be present so scripts will
+run with default params. A module is a .ps1 script starting with Get-.
+#>
+Param(
+    [Parameter(Mandatory=$True,Position=0)]
+        [String]$ModulePath
+)
+    Write-Debug "Entering $($MyInvocation.MyCommand)"
+    $Error.Clear()
+    # ToDo: There should probably be some error handling in this function.
+
+    Write-Debug "`$ModulePath is ${ModulePath}."
+
+    # User may have passed a full path to a specific module, possibly with an argument
+    $ModuleScript = ($ModulePath -split " ")[0]
+    $ModuleArgs   = @($ModulePath -split [regex]::escape($ModuleScript))[1].Trim()
+
+    $Modules = $FoundModules = @()
+    # Need to maintain the order for "order of volatility"
+    $ModuleHash = New-Object System.Collections.Specialized.OrderedDictionary
+
+    if (!(ls $ModuleScript | Select-Object -ExpandProperty PSIsContainer)) {
```

Comment:
From the contributing.md: 

> Do error on the side of verbosity rather than compactness, avoid aliases.

In general, we've mostly avoided the use of aliases in Kansa. It makes the code longer, but it also makes it easier to people to understand without having to look up aliases. Using ls is probably fine, but there are other aliases that aren't obvious to all developers.